ivre/db/mongo.py

```diff
@@ -4478,3 +4504,1088 @@ def get_masters(self):
+                    field_name = reverse_project_fields[cur_field]
+                else:
+                    field_name = "field%s" % index
+                    project_fields[field_name] = cur_field
+                    reverse_project_fields[cur_field] = field_name
+                new_field_name = "$%s" % field_name
+                # _id group
+                if i == 0:
+                    group_fields[i][field_name] = new_field_name
+                # collect group
+                else:
+                    group_fields[i][field_name] = {'$push': new_field_name}
+                index += 1
+        # Add sum projection if sum_fields are provided
+        if sum_fields:
+            project_fields['_sum'] = {
+                '$add': ['$%s' % field for field in internal_fields[2]]}
+
+        pipeline.append({'$project': project_fields})
+
+        # Group stage
+        group = group_fields[1]
+        group['_id'] = group_fields[0]
+        group['_count'] = {'$sum': '$_sum' if sum_fields else 1}
+        pipeline.append({'$group': group})
+
+        pipeline.append({"$sort": {"_count": 1 if least else -1}})
+
+        if skip is not None:
+            pipeline.append({"$skip": skip})
+        if limit is not None:
+            pipeline.append({"$limit": limit})
+
+        res = self.db[self.columns[self.column_flow]].aggregate(pipeline)
+        for entry in res:
+            # Translate again the collected fields
+            ext_entry = {}
+            for key, value in entry.items():
+                if key in project_fields:
+                    ext_entry[project_fields[key][1:]] = value
+                else:
+                    ext_entry[key] = value
+            # Translate again the aggr fields
+            ext_entry['_id'] = {}
+            for key, value in entry['_id'].items():
+                if key in project_fields:
+                    ext_entry['_id'][project_fields[key][1:]] = value
+                else:
+                    ext_entry['_id'][key] = value
+            # apply internal2ip to addr results
+            for addr_field in ['src_addr', 'dst_addr']:
+                addr0, addr1 = (addr_field + '_0', addr_field + '_1')
+                addr = addr_field[:3] + '.addr'
+                # Apply in aggregate fields
+                if addr0 in ext_entry['_id'] and addr1 in ext_entry['_id']:
+                    ext_entry['_id'][addr] = self.internal2ip(
+                        (ext_entry['_id'][addr0], ext_entry['_id'][addr1]))
+                    ext_entry['_id'].pop(addr0)
+                    ext_entry['_id'].pop(addr1)
+                # Apply in collected fields
+                if addr0 in ext_entry and addr1 in ext_entry:
+                    ext_entry[addr] = []
+                    for i in range(len(ext_entry[addr0])):
+                        ext_entry[addr].append(self.internal2ip(
+                            [ext_entry[addr0][i], ext_entry[addr1][i]]))
```

Comment:
Use `.pop()`

Reply:
`pop` can not be used here since `ext_entry[addrX]` is read in the for loop and is deleted after this loop. If I replace 
```Python
ext_entry[addr].append(self.internal2ip(
                            [ext_entry[addr0][i], ext_entry[addr1][i]]))
```
by 
```Python
ext_entry[addr].append(self.internal2ip(
                            [ext_entry.pop(addr0)[i], ext_entry.pop(addr1)[i]]))
```
I'll get an AttributeError at the second iteration.